documentation/bolt_installing_modules.md

```diff
@@ -329,44 +329,136 @@ module-install:
-
-If you want to install a module to an existing project, use a Puppetfile. This
-method does not automatically resolve module dependencies. If the module you're
-installing requires other modules, make sure you add the required modules to
-your Puppetfile together with the module you're installing.
-
-> **Before you begin**
->
-> - In your Bolt project directory, create a file named `Puppetfile`. 
-> - Add any modules stored locally in `modules/` to the list. For example, 
->   ```puppet
->     mod 'my_awesome_module', local: true
->   ```
->
->   **Bolt deletes any content in `modules/` that is not listed in your
->   Puppetfile.** If you want to keep the content, but you don't want to manage
->   it with the Puppetfile, move the content to a `site-modules` directory in
->   your project.
-
-To install a module:
-   1.  Open Puppetfile in a text editor and add the modules and versions that
-       you want to install. If the modules have dependencies, list those as
-       well. For example:
-       ```puppet
-       # Modules from the Puppet Forge.
-       mod 'puppetlabs-apache', '4.1.0'
-       mod 'puppetlabs-postgresql', '5.12.0'
-       mod 'puppetlabs-puppet_conf', '0.3.0'
-
-       # Modules from a Git repository.
-       mod 'puppetlabs-haproxy', git: 'https://github.com/puppetlabs/puppetlabs-haproxy.git', ref: 'master'
-       ```   
-   1. Run the `bolt puppetfile install` command. Bolt installs modules to the
-      first directory in the `modulepath` setting. By default, this is the
-      `modules/` subdirectory inside the Bolt project directory. To override
-      this location, update the `modulepath` setting in your [project
-      configuration file](bolt_project_reference.md).
+## Manually manage a project's modules
+
+If Bolt can't resolve your module dependencies, you must manage your project's
+Puppetfile manually and use Bolt to install the modules listed in the Puppetfile
+without resolving dependencies. The process for manually managing your modules
+uses the new `module` subcommand, and replaces the now deprecated `puppetfile`
+subcommand.
+
+The most common scenario where Bolt can't resolve module dependencies is when
+a project includes git modules that are in a repository other than a public
+GitHub repository. If your project includes this type of module, you must
+manually manage your project's Puppetfile.
+
+To manually manage a project's Puppetfile and install modules without resolving
+dependencies, follow these steps:
+
+1. Set the `modules` configuration option to an empty array. For example:
+
+   ```yaml
+   # bolt-project.yaml
+   name: myproject
+   modules: []
+   ```
+
+1. Create a file named `Puppetfile` in your project directory and add the
+   modules you want to install, including each module's dependencies, to
+   the Puppetfile. For example:
+
+   ```ruby
+   # Modules from a private git repository
+   mod 'private-module', git: 'https://github.com/bolt-user/private-module.git', ref: 'main'
```

Comment:
Hm, the error message for this case is actually a little confusing:
```
lucy ~/githubs/bolt/project ((eb3edb98...)) $ be bolt module install --force
Installing project modules

Missing metadata.json at https://github.com/puppetlabs/teams-ui-testing-module. This is not a valid module.
```
I know it's upstream, so I'll make a ticket for it.